src/views/components/shared/Input/component.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { Form, Input as AntInput } from 'antd'
+import { EyeTwoTone, EyeInvisibleOutlined } from '@ant-design/icons'
+
+import { propTypes, defaultProps } from './types'
+
+export const iconRender = (visible) =>
+  visible ? <EyeTwoTone /> : <EyeInvisibleOutlined />
+
+const Input = ({
+  placeholder,
+  validateStatus,
+  validationMessage,
+  type,
+  value,
+  onChange,
+  onBlur,
+  name,
+  prefix
+}) => (
+  <Form.Item
+    validateStatus={validateStatus}
+    help={validationMessage}
+  >
+    {type === 'password' ? (
+      <AntInput.Password
```

Comment:
You can simplify this by passing `as` prop with component which should be rendered.

```
defaultProps = {
  as: 'AntInput'
}
```

```
<Field
  as={AntInput.Password}
```

Reply:
I work on it, good point